### PR TITLE
feat(super user panel): add a section for Data Sources to the WS page

### PR DESCRIPTION
**Pull request by fontanierh**
- you can see the list of data sources (managed first, then non-managed)
- you can see how many docs they have
- you can see the last successful sync time
- you can delete them
    - I fixed a pretty sad bug on `connectors` where the `Connector` object's deletion wasn't actually committed
    - Currently, we do not scrub the GCP data -- wondering if we should cc @spolu

### CODE REVIEW

**Comment by fontanierh** [comment]
Screenshot 2023-08-02 at 19 12 16
Screenshot 2023-08-02 at 19 12 09
Screenshot 2023-08-02 at 19 12 04

**Comment by spolu** [comment]
- We def should scrub GCP data I think 👍 delete really is the last step of the funnel.
- what happens if we downgrade without deleting data sources?
- Can we pause DataSources before deleting then which was identified by @gabhubert as the first step of ending the free trial?

**Comment by fontanierh** [comment]
> We def should scrub GCP data I think 👍 delete really is the last step of the funnel.

Adding 👍 

> what happens if we downgrade without deleting data sources?

Behaviour is ISO with the existing CLI

> Can we pause DataSources before deleting then which was identified by @gabhubert as the first step of ending the free trial?

I think we can explore that, but currently:
- only github and notion support the stop/resume interface
- we don't have an existing way to exclude a datasource from retrievals
- overall, I think "ending free trial" requires a lot more framing

File: front/pages/api/poke/workspaces/[wId]/data_sources/[name]/index.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      }
+
+      await CoreAPI.deleteDataSource({
+        projectId: dustAPIProjectId,
+        dataSourceName: dataSource.name,
+      });
+
+      await dataSource.destroy();
+
+      // TODO SHOULD WE SCRUB ?!
```
[the remark this answers is not in the document]
Comment (PopDaph) [reply front/pages/api/poke/workspaces/[wId]/data_sources/[name]/index.ts:115]: https://github.com/dust-tt/dust/pull/1075

File: front/pages/poke/[wId]/index.tsx [hunk 2]
```diff
@@ -111,34 +215,76 @@ const WorkspacePage = ({
...
+                <p className="mb-2 text-sm text-gray-600">
+                  {ds.connectorProvider ? "Managed - " : "Non-managed - "}{" "}
+                  <span className="font-medium">
+                    {documentCounts[ds.name] ?? 0}
+                  </span>{" "}
+                  documents
+                </p>
+                {dataSourcesSynchronizedAgo[ds.name] && (
+                  <p className="mb-2 text-sm text-gray-600">
+                    Synchronized {dataSourcesSynchronizedAgo[ds.name]} ago
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/pages/poke/[wId]/index.tsx:281]: as per irl, will do in separate PR -- we want essentially the same status the admin has on their DS settings page

File: front/pages/poke/[wId]/index.tsx [hunk 3]
```diff
@@ -111,34 +215,76 @@ const WorkspacePage = ({
...
+              <span className="mx-2 w-1/3">
+                <p className="mb-4 text-sm text-warning ">
+                  Delete managed data sources before downgrading.
+                </p>
+              </span>
+            )}
+          </div>
+          <div className="mx-2 w-1/3">
+            <h2 className="text-md mb-4 font-bold">Data Sources:</h2>
+            {dataSources.map((ds) => (
```
Comment (PopDaph) [inline front/pages/poke/[wId]/index.tsx:259]: NIT: empty state if there are no datasources? just a sentence

**Review by PopDaph — APPROVED** [review]

**Pull request merged**